[16.0][OU-ADD] website_sale_loyalty_delivery: Migration to version 16.0

pilarvargas-tecnativa opened the pull request:
cc @tecnativa TT49821

@pedrobaeza @chienandalu please review

pilarvargas-tecnativa committed:
[OU-ADD] website_sale_loyalty_delivery: Migration to version 16.0

TT49821

pedrobaeza commented:
This is not `[MIG]`, but `[OU-ADD]`.

pedrobaeza commented:
/ocabot migration website_sale_loyalty_delivery

<dependency>Depends on : 

- sale_loyalty_delivery : #4506</dependency>

pilarvargas-tecnativa commented:
@pedrobaeza done! thanks!

pedrobaeza commented on docsource/modules150-160.rst:
```diff
@@ -908,7 +908,7 @@ Module coverage 15.0 -> 16.0
 | |new| website_sale_loyalty                      |                      |                                                 |
 +-------------------------------------------------+----------------------+-------------------------------------------------+
-| |new| website_sale_loyalty_delivery             |                      |                                                 |
+| |new| website_sale_loyalty_delivery             | Nothing to do        |                                                 |
```
```suggestion
| |new| website_sale_loyalty_delivery             | Done                 | Renamed from website_sale_coupon_delivery       |
```

pedrobaeza replied:
https://github.com/OCA/OpenUpgrade/blob/3061ff30d09754ac158c860538165bffdb3672f2/openupgrade_scripts/apriori.py#L14

pedrobaeza replied:
And you have to put the contrary one (`Renamed to...`) on `website_sale_coupon_delivery` del entry.

pilarvargas-tecnativa replied:
Done!

pedrobaeza commented on docsource/modules150-160.rst:
```diff
@@ -894,7 +894,7 @@ Module coverage 15.0 -> 16.0
 | |del| website_sale_coupon                       |                      |                                                 |
 +-------------------------------------------------+----------------------+-------------------------------------------------+
-| |del| website_sale_coupon_delivery              |                      |                                                 |
+| |del| website_sale_coupon_delivery              |                      |Renamed to website_sale_loyalty_delivery         |
```
```suggestion
| |del| website_sale_coupon_delivery              | Done                  |Renamed to website_sale_loyalty_delivery         |
```

pedrobaeza commented on docsource/modules150-160.rst:
```diff
@@ -908,7 +908,7 @@ Module coverage 15.0 -> 16.0
 | |new| website_sale_loyalty                      |                      |                                                 |
 +-------------------------------------------------+----------------------+-------------------------------------------------+
-| |new| website_sale_loyalty_delivery             |                      |                                                 |
+| |new| website_sale_loyalty_delivery             | Nothing to do        |Renamed from website_sale_coupon_delivery        |
```
```suggestion
| |new| website_sale_loyalty_delivery             | Nothing to do        | Renamed from website_sale_coupon_delivery       |
```

pedrobaeza reviewed (approved):
/ocabot merge nobump

pilarvargas-tecnativa committed:
[OU-IMP] sale_loyalty_delivery: Adding information to the module coverage table

pilarvargas-tecnativa committed:
[OU-IMP] loyalty: Adding information to the module coverage table

pilarvargas-tecnativa committed:
[OU-IMP] sale_loyalty: Adding information to the module coverage table

pilarvargas-tecnativa committed:
[OU-IMP] website_sale_loyalty: Adding information to the module coverage table

pilarvargas-tecnativa committed:
[OU-FIX] sale_loyalty: Remove script that removes constraint

This script is not necessary since both the table and the constraint
are new in v16, it is only necessary to apply this restriction to the
data dump in the new table by adding a condition that contemplates it.

pedrobaeza commented on openupgrade_scripts/scripts/sale_loyalty/16.0.1.0/post-migration.py:
```diff
@@ -80,15 +80,21 @@ def _generate_sale_order_coupon_points(env):
+            lc.write_date
+        FROM loyalty_card lc
+        WHERE lc.order_id IS NOT NULL
+        AND NOT EXISTS (
```
Instead of this, do a more performant `LEFT JOIN` with the table, and check if `socp.id IS NULL` as the WHERE condition.

pilarvargas-tecnativa replied:
thanks!
